contracts/handlers/GenericHandler.sol

```diff
@@ -139,20 +150,32 @@ contract GenericHandler is IGenericHandler {
         {metaData} is expected to consist of needed function arguments.
      */
     function deposit(bytes32 resourceID, uint8 destinationChainID, uint64 depositNonce, address depositer, bytes calldata data) external onlyBridge {
-        uint      lenMetadata;
+        uint256      lenMetadata;
         bytes memory metadata;
 
-        lenMetadata = abi.decode(data, (uint));
+        lenMetadata = abi.decode(data, (uint256));
         metadata = bytes(data[32:32 + lenMetadata]);
 
         address contractAddress = _resourceIDToContractAddress[resourceID];
+        uint256 depositerOffset = _contractAddressToDepositFunctionDepositerOffset[contractAddress];
```

Comment:
I just want to make sure my understanding of this is correct: the new format of the data being passed into the function is as follows:

```
len(data)                              uint256     bytes  0  - 32
data                                      bytes       bytes  32 - END
--------------------------------------------
Within data:
0 - 0 + depositorOffset         encoded arguments (variable length)
depositorOffset - END         depositorAddress      (20 bytes long)


```

Reply:
since I think we're passing in the entire contents of data into the function call as function arguments on line 176, is there any situation where this might cause an issue?

Reply:
nevermind, chatted about this offline